### DISCUSSION TITLE
k3s.service from install script is wrong?

### DISCUSSION THREAD

**Discussion by stratus-ss** [k3s-io/k3s#6275] [Q&A]
**Environmental Info:**
K3s Version: 

```
k3s -v
k3s version v1.24.6+k3s1 (a8e0c66d)
go version go1.18.6
```

Node(s) CPU architecture, OS, and Version: 

Single node on an AMD EPYC cpu (x86_64)

CentOS Stream 9
```
uname -a
Linux bastion.stratus.lab 5.14.0-171.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Oct 1 12:56:27 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:

Single Node install with the script in the [quick start](https://docs.k3s.io/quick-start) 

**Describe the bug:**

Running the script does not allow the k3s to start. It appears to be a problem with the service file.

**Steps To Reproduce:**

- Installed K3s:  

No customizations have been done. Simply curl the script in the on the quick-start page.

**Expected behavior:**

Cluster initializes and the service starts

**Actual behavior:**

During the script running the following output is observed:

```
[INFO]  env: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
Created symlink /etc/systemd/system/multi-user.target.wants/k3s.service → /etc/systemd/system/k3s.service.
[INFO]  systemd: Starting k3s
Job for k3s.service failed because the control process exited with error code.
See "systemctl status k3s.service" and "journalctl -xeu k3s.service" for details.

```

Investigating the logs I see the following:

```
Oct 14 13:29:18 bastion.stratus.lab sh[37322]: + /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
Oct 14 13:29:18 bastion.stratus.lab sh[37323]: Failed to get unit file state for nm-cloud-setup.service: No such file or directory
Oct 14 13:29:18 bastion.stratus.lab k3s[37326]: Incorrect Usage: flag provided but not defined: -hg
```

The service file has the following content:

```
[Unit]
Description=Lightweight Kubernetes
Documentation=https://k3s.io
Wants=network-online.target
After=network-online.target

[Install]
WantedBy=multi-user.target

[Service]
Type=notify
EnvironmentFile=-/etc/default/%N
EnvironmentFile=-/etc/sysconfig/%N
EnvironmentFile=-/etc/systemd/system/k3s.service.env
KillMode=process
Delegate=yes
# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNOFILE=1048576
LimitNPROC=infinity
LimitCORE=infinity
TasksMax=infinity
TimeoutStartSec=0
Restart=always
RestartSec=5s
ExecStartPre=/bin/sh -xc '! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service'
ExecStartPre=-/sbin/modprobe br_netfilter
ExecStartPre=-/sbin/modprobe overlay
ExecStart=/usr/local/bin/k3s \
    server \
	'-hg' \
```

When I investigate the the help on the binary, i am unable to determine what `-hg` is supposed to do. When I look at the github page the [k3s.service](https://github.com/k3s-io/k3s/blob/master/k3s.service) file looks different from the one that the script installed

**Comment by brandond** [comment]
Any extra args that you provide to the shell command when running the installer will be appended to the K3s command in the systemd unit. Did you perhaps happen to run something like:
```bash
curl -sfL https://get.k3s.io | sh - -hg
```

**Comment by stratus-ss** [comment]
I did not. Looking at my history I did this (in line with the quick start page):

```
911  curl -sfL https://get.k3s.io | sh -
922  curl -sfL https://get.k3s.io -o install-k3s.sh
```

However my history has rolled over now, so I suppose it is possible that the `install-k3s.sh ` file was launched in such a fashion. It's a plausible explanation

**Reply by brandond** [reply]
`INSTALL_K3S_EXEC` is passed through in the same fashion, but it seems less likely that you'd have that set. Either way, it came from something on your side. You can see the unit template here: https://github.com/k3s-io/k3s/blob/master/install.sh#L802-L803

**Comment by BasToTheMax** [comment]
I have the same problem. Are there any fixes found (yet)?

**Reply by brandond** [reply]
As described above, there is nothing in the install script's systemd unit template that's doing this. It is passing through something from your installation command. If you can provide any information on what specifically you did to cause this, we might be able to help you.
